### PR TITLE
Bug 1859393 - Add CFR when the first cookie banner gets cleared.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1859393

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt [hunk 1]
```diff
@@ -325,6 +353,72 @@ class BrowserToolbarCFRPresenter(
         }
     }
 
+    @OptIn(ExperimentalComposeUiApi::class)
```
Comment (iorgamgabriel) [inline fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt:356]: You should remove this line @OptIn(ExperimentalComposeUiApi::class).

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt [hunk 2]
```diff
@@ -325,6 +353,72 @@ class BrowserToolbarCFRPresenter(
...
+                    CFRPopup.IndicatorDirection.UP
+                } else {
+                    CFRPopup.IndicatorDirection.DOWN
+                },
+            ),
+            onDismiss = {
+                when (it) {
+                    true -> {
+                        TrackingProtection.tcpCfrExplicitDismissal.record(NoExtras())
+                        settings.shouldShowTotalCookieProtectionCFR = false
```
Comment (iorgamgabriel) [inline fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt:382]: I think here it should be settings.shouldShowCookieBannersCFR = false  ,
If it is needed for the CFR to be shown only once.

File: fenix/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -498,6 +498,8 @@
     <string name="reduce_cookie_banner_dialog_snackbar_text" moz:RemovedIn="121" tools:ignore="UnusedResources">You’ll see fewer cookie requests</string>
     <!-- Change setting text button, for the cookie banner re-engagement dialog -->
     <string name="reduce_cookie_banner_dialog_change_setting_button" moz:RemovedIn="121" tools:ignore="UnusedResources">Allow</string>
+    <string name="cookie_banner_cfr_title">Firefox just refused cookies for you</string>
+    <string name="cookie_banner_cfr_message">Less distractions, less cookies tracking you on this site.</string>
```
Comment (iorgamgabriel) [inline fenix/app/src/main/res/values/strings.xml:502]: Please add comments for the strings .

File: fenix/app/src/test/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenterTest.kt [hunk 4]
```diff
@@ -69,6 +71,42 @@ class BrowserToolbarCFRPresenterTest {
...
+
+        assertNotNull(presenter.scope)
+
+        browserStore.dispatch(CookieBannerAction.UpdateStatusAction(privateTab.id, HANDLED))
+            .joinBlocking()
+
+        verify(exactly = 1) { presenter.showCookieBannersCFR() }
+
+//        browserStore.dispatch(ContentAction.UpdateProgressAction(customTab.id, 33)).joinBlocking()
+//        verify(exactly = 0) { presenter.showTcpCfr() }
```
Comment (iorgamgabriel) [inline fenix/app/src/test/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenterTest.kt:104]: This test is failing. I don't know if we need these commented lines.

File: android-components/components/compose/cfr/src/main/java/mozilla/components/compose/cfr/CFRPopup.kt [hunk 5]
```diff
@@ -155,7 +155,7 @@ class CFRPopup(
         /**
          * Default width for all CFRs.
          */
-        internal const val DEFAULT_WIDTH = 335
+        const val DEFAULT_WIDTH = 335
```
Comment (iorgamgabriel) [inline android-components/components/compose/cfr/src/main/java/mozilla/components/compose/cfr/CFRPopup.kt:158]: Do we need to remove internal ?

**Comment by iorgamgabriel** [comment]
Detekt and ktlint are failing.

**Review by iorgamgabriel — CHANGES_REQUESTED** [review]
Detekt and ktlint are failing.

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt (hunk 1 above)
Comment (Amejia481) [reply fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt]: Addressed!

File: fenix/app/src/main/res/values/strings.xml (hunk 3 above)
Comment (Amejia481) [reply fenix/app/src/main/res/values/strings.xml:504]: Addressed!

File: android-components/components/compose/cfr/src/main/java/mozilla/components/compose/cfr/CFRPopup.kt (hunk 5 above)
Comment (Amejia481) [reply android-components/components/compose/cfr/src/main/java/mozilla/components/compose/cfr/CFRPopup.kt]: Yes, addressed!

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt (hunk 2 above)
Comment (Amejia481) [reply fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarCFRPresenter.kt]: Addressed!

**Review by iorgamgabriel — APPROVED** [review]

**Pull request merged**